jbpm-test-coverage/src/test/java/org/jbpm/test/functional/log/ProcessInstanceLogArchiveTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.test.functional.log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.EntityManager;
+
+import org.jbpm.process.audit.JPAAuditLogService;
+import org.jbpm.process.audit.NodeInstanceLog;
+import org.jbpm.test.JbpmTestCase;
+import org.junit.Assert;
+import org.junit.Test;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.process.ProcessInstance;
+
+/**
+ * TODO: Add deleteLogsByDate, deleteLogsByDateRange (however it is a bit redundant).
+ */
+public class ProcessInstanceLogArchiveTest extends JbpmTestCase {
+
+    private static final String HELLO_WORLD_PROCESS =
+            "org/jbpm/test/functional/common/HelloWorldProcess1.bpmn";
+    private static final String HELLO_WORLD_PROCESS_ID =
+            "org.jbpm.test.functional.common.HelloWorldProcess1";
+
+
+    private JPAAuditLogService auditService;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        auditService = new JPAAuditLogService(getEmf());
+        auditService.clear();
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        try {
+            auditService.clear();
+            auditService.dispose();
+        } finally {
+            super.tearDown();
+        }
+    }
+
+    @Test
+    public void checkEndDateArchived() {
+        KieSession kieSession = createKSession(HELLO_WORLD_PROCESS);
+
+        startProcess(kieSession, HELLO_WORLD_PROCESS_ID, 2);
+
+        EntityManager em = getEmf().createEntityManager();
+        List<NodeInstanceLog> logs = em.createQuery("SELECT o FROM NodeInstanceLog o", NodeInstanceLog.class).getResultList();
```

Comment:
maybe it might be worth adding some other audit tables in this test

Reply:
added